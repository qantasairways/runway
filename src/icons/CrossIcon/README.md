```js
<CrossIcon width="50" height="50" />
```
