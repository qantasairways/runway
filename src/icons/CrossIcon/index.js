import React from 'react';
import PropTypes from 'prop-types';
import { generate } from 'shortid';

function CrossIcon({ height, width, className }) {
  const pathId = generate();
  const maskId = generate();

  return (
    <svg
      width={width}
      height={height}
      className={className}
      focusable="false"
      viewBox="-0.5 -0.5 24 24"
    >
      <defs>
        <path
          id={pathId}
          d="M12.893.302a.996.996 0 0 0-1.41 0l-4.89 4.88-4.89-4.89a.996.996 0 1 0-1.41 1.41l4.89 4.89-4.89 4.89a.996.996 0 1 0 1.41 1.41l4.89-4.89 4.89 4.89a.996.996 0 1 0 1.41-1.41l-4.89-4.89 4.89-4.89c.38-.38.38-1.02 0-1.4z"
        />
      </defs>
      <g fillRule="evenodd" transform="translate(5 5)">
        <mask id={maskId}>
          <use xlinkHref={`#${pathId}`} />
        </mask>
        <use fillRule="nonzero" xlinkHref={`#${pathId}`} />
        <g mask={`url(#${maskId})`}>
          <path d="M-5-5h24v24H-5z" />
        </g>
      </g>
    </svg>
  );
}

CrossIcon.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string,
  className: PropTypes.string
};

CrossIcon.defaultProps = {
  height: '24',
  width: '24',
  className: ''
};

export default CrossIcon;
