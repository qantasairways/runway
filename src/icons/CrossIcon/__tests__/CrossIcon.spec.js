import React from 'react';
import { shallow } from 'enzyme';
import shortid from 'shortid';

import CrossIcon from '..';

jest.mock('shortid');

describe('CrossIcon', () => {
  it('should render', () => {
    shortid.generate.mockReturnValue('1a2b3c');
    const component = shallow(
      <CrossIcon width="36px" height="36px" className="my-favourite-icon" />
    );

    expect(component).toMatchSnapshot();
  });

  it('should set the width', () => {
    const component = shallow(<CrossIcon width="36px" />);
    const result = component.find('svg').props().width;

    expect(result).toEqual('36px');
  });

  it('should set the height', () => {
    const component = shallow(<CrossIcon height="36px" />);
    const result = component.find('svg').props().height;

    expect(result).toEqual('36px');
  });

  it('should set the class name', () => {
    const component = shallow(<CrossIcon className="my-favourite-icon" />);
    const result = component.find('svg').props().className;

    expect(result).toEqual('my-favourite-icon');
  });
});
