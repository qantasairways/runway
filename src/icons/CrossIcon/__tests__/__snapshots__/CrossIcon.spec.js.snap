// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrossIcon should render 1`] = `
<svg
  className="my-favourite-icon"
  focusable="false"
  height="36px"
  viewBox="-0.5 -0.5 24 24"
  width="36px"
>
  <defs>
    <path
      d="M12.893.302a.996.996 0 0 0-1.41 0l-4.89 4.88-4.89-4.89a.996.996 0 1 0-1.41 1.41l4.89 4.89-4.89 4.89a.996.996 0 1 0 1.41 1.41l4.89-4.89 4.89 4.89a.996.996 0 1 0 1.41-1.41l-4.89-4.89 4.89-4.89c.38-.38.38-1.02 0-1.4z"
      id="1a2b3c"
    />
  </defs>
  <g
    fillRule="evenodd"
    transform="translate(5 5)"
  >
    <mask
      id="1a2b3c"
    >
      <use
        xlinkHref="#1a2b3c"
      />
    </mask>
    <use
      fillRule="nonzero"
      xlinkHref="#1a2b3c"
    />
    <g
      mask="url(#1a2b3c)"
    >
      <path
        d="M-5-5h24v24H-5z"
      />
    </g>
  </g>
</svg>
`;
